src/boot/bootctl.c

```diff
@@ -1647,6 +1653,12 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_arch_all = true;
                         break;
 
+                case ARG_EFI_LABEL:
+                        r = free_and_strdup(&arg_efi_label, optarg);
+                        if (r < 0)
+                                return log_oom();
+                        break;
```

Comment:
I am not sure what kind of string is supported, but I think it is better to check the string at least with `string_is_safe()`. Also, it is better to check the length.

Reply:
Looking at the UEFI specification there is no mention of an upper limit for the name of the entry (referred to as "Description" in the spec): https://uefi.org/sites/default/files/resources/UEFI_Spec_2_9_2021_03_18.pdf (page 74)
Longest string I bothered testing was 16384 times the letter x as a label:
![image](https://user-images.githubusercontent.com/5379596/184509785-da7d13d3-aacb-4bde-affd-2b5c539ba94a.png)

Nonetheless I added the checks as suggested. See manual tests in initiall PR message.

Reply:
Wow...

Reply:
You should be *very* careful testing this. You could end up bricking your device writing too much data into nvram. You should delete that var and do a cold boot or two to make sure the firmware had a chance to reclaim that space.

Reply:
I removed it again and my PC booted fine this morning :)